src/main/java/Duke.java

```diff
@@ -1,10 +1,84 @@
+import java.util.Scanner;
+
 public class Duke {
+    private static final int MAX_TASKS = 100;
+    private static final String DIVIDER_LINE = "____________________________________________________________";
+    
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // Display the welcome message
+        displayWelcomeMessage();
+        
+        Scanner scanner = new Scanner(System.in);
+        Task[] tasks = new Task[MAX_TASKS];
+        int taskCount = 0;
+        
+        String inputLine;
+        System.out.println("What can I do for you?\n" + DIVIDER_LINE);
+
+        do {
+            inputLine = scanner.nextLine();
+            System.out.println(DIVIDER_LINE);
+            if (inputLine.equals("list")) {
+                displayTaskList(tasks, taskCount);
+            } else if (inputLine.startsWith("mark ")) {
+                markTaskAsDone(tasks, inputLine, taskCount);
+            } else if (inputLine.startsWith("unmark ")) {
+                markTaskAsNotDone(tasks, inputLine, taskCount);
+            } else if (!inputLine.equals("bye")) {
+                addTask(tasks, inputLine, taskCount);
+                taskCount++;
+            }
+            System.out.println(DIVIDER_LINE);
+        } while (!inputLine.equals("bye"));
+
+        System.out.println("Bye. Hope to see you again soon!");
+        scanner.close();
+    }
+
+    private static void displayWelcomeMessage() {
+        String logo = " ____  _    _  ____ _____ ___  ____  \n"
```

Comment:
I would recommend to make this a static constant outside the method and capitalize the name.